programs/temp/src/instructions/create_nft.rs

```diff
@@ -11,30 +14,18 @@ use anchor_spl::token::{Mint, Token};
 pub struct MintNFT<'info> {
     #[account(mut)]
     pub mint_authority: Signer<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
-    #[account(
-        init_if_needed,
-        payer = payer,
-        space = Mint::LEN
-    )]
+    #[account(mut)]
     pub mint: Account<'info, Mint>,
-    // #[account(mut)]
     pub token_program: Program<'info, Token>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     #[account(mut)]
-    pub metadata: UncheckedAccount<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
+    pub metadata: AccountInfo<'info>,
     #[account(mut)]
     pub token_account: UncheckedAccount<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     pub token_metadata_program: UncheckedAccount<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     #[account(mut)]
     pub payer: AccountInfo<'info>,
     pub system_program: Program<'info, System>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     pub rent: AccountInfo<'info>,
-    /// CHECK: This is not dangerous because we don't read or write from this account
     #[account(mut)]
     pub master_edition: UncheckedAccount<'info>,
```

Comment:
Can we use metaplex account types here? If not, just use UncheckedAccount, but make it clear that metaplex is doing the checking via CPI